x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/job_map.tsx

```diff
@@ -37,102 +37,78 @@ ${theme.euiColorLightShade}`,
   marginTop: 0,
 };
 
-export const JobMapTitle: React.FC<{ analyticsId?: string; modelId?: string }> = ({
-  analyticsId,
-  modelId,
-}) => (
-  <EuiTitle size="xs">
-    <span>
-      {analyticsId
-        ? i18n.translate('xpack.ml.dataframe.analyticsMap.analyticsIdTitle', {
-            defaultMessage: 'Map for analytics ID {analyticsId}',
-            values: { analyticsId },
-          })
-        : i18n.translate('xpack.ml.dataframe.analyticsMap.modelIdTitle', {
-            defaultMessage: 'Map for trained model ID {modelId}',
-            values: { modelId },
-          })}
-    </span>
-  </EuiTitle>
-);
-
-interface GetDataObjectParameter {
-  id: string;
-  type: string;
-}
-
 interface Props {
   analyticsId?: string;
   modelId?: string;
 }
 
 export const JobMap: FC<Props> = ({ analyticsId, modelId }) => {
-  const [elements, setElements] = useState<cytoscape.ElementDefinition[]>([]);
-  const [nodeDetails, setNodeDetails] = useState({});
-  const [error, setError] = useState(undefined);
+  const [itemsDeleted, setItemsDeleted] = useState<boolean>(false);
```

Comment:
Can you add a comment how this is used and triggered? I see code that's able to set it to `true` but it will never return to a state of `false` is that correct?

Reply:
That's correct - `itemsDeleted` gets reset to `false` because the the `Controls` component uses the `updateElements` prop to remove the deleted nodes from the map - so it causes `JobMap` to re-render with the new elements. I can add a comment to clarify. 👍